Tavisca.Bootcamp.LanguageBasics.Exercise1/Program.cs

```diff
@@ -4,26 +4,121 @@ namespace Tavisca.Bootcamp.LanguageBasics.Exercise1
 {
     class Program
     {
-        static void Main(string[] args)
+        static void Main(string [] args)
         {
             Test("42*47=1?74", 9);
             Test("4?*47=1974", 2);
             Test("42*?7=1974", 4);
             Test("42*?47=1974", -1);
             Test("2*12?=247", -1);
-            Console.ReadKey(true);
+            Test("42*47=1??4",-1);
+          
         }
-
-        private static void Test(string args, int expected)
-        {
+          private static void Test(string args, int expected)
+        {
             var result = FindDigit(args).Equals(expected) ? "PASS" : "FAIL";
-            Console.WriteLine($"{args} : {result}");
+           Console.WriteLine($"{args} : {result}");
         }
 
-        public static int FindDigit(string equation)
-        {
-            // Add your code here.
-            throw new NotImplementedException();
+        public static int FindDigit(string equation)
+        { 
+           var val= checkString(equation);
+            if(val == -1) return -1;
+            
+        string [] str = equation.Split(new char[] {'*','='} );
+           int num,mul=1,ind=0,result1;
+           string str1="";
+           int ind1= equation.IndexOf('?');
+           if(ind1 == -1) return -1;
+           int ind2 =equation.IndexOf('=');
+           if(ind1>ind2)
+           {
+            foreach(var c in str)
+            {
+            
+            num = getValue(c);
+            if(num>0)
+            mul *= num;
+            else{
+                 str1=c;
+               ind= str1.IndexOf('?');
+            }
+            }
+            string s = mul.ToString();
+            if(s.Length != str1.Length)
+            {
+                return -1;
+            }
+            return (s[ind]-'0');
+           }
+
+           else if(ind1 < ind2){
+                int divisor = getValue(str[2]);
+                int a = getValue(str[0]);
+                int b =getValue(str[1]);
+                if(a>0)
+                {
+                    if(divisor%a == 0) result1 = divisor/a;
+                    else return -1;
+                    ind= str[1].IndexOf('?');
+                    str1=str[1];

```

Comment:
keep str[0] and others into strA, strB and strC variables as they are getting used repeatedly